### DISCUSSION TITLE
Discussion Categories, Formats & Labels

### DISCUSSION THREAD

**Discussion by simonhkswan** [community/community#5263] [Discussions]
Firstly, just wanted say how fantastic of a feature this is, i'm so excited to see how this will impact day-to-day development and planning. There aren't many project planning applications which really support the initial "draft"/discussion stage of an idea quite like this. 🚀 

The thing that's bothering me at the moment is the pretty overlapping usage of **Categories**, **Discussion Formats**, 
 & **Labels**.
 
###  Currently there are 5 default "Categories" split amongst 3 "Discussion Formats" all of which can have any labels. I've found that unnecessarily complicated.

- Open ended discussion
  - 💬 General
  - 💡 Ideas
  - 🙌 Show and tell
- Question / Answer
  - 🙏 Q&A 
- Announcement
  - 📣 Announcements

What I've felt is that when going to make a discussion, I'm repeatedly finding myself unsure about which category my discussion belongs to. *"I've got some general ideas that I want to show and tell and have a few questions to ask... what Category do I pick?!"*

### Why don't we just use labels to organize discussions into different topics? It's what we do for Issues and PRs afterall.

Discussions are supposed to facilitate the development of incomplete thoughts & ideas. They don't *need* to be categorised into arbitrary types before conversation takes place. If i want to discuss something, It should be as simple as writing the title and then setting how others are able to reply. I can apply labels afterwards in any combination that is decided by the repo's guidelines after.

### Discussion Formats look great, these ARE mutually exclusive and need to be decided on before starting the discussion. They could replace Categories.

Maybe this should be used directly instead of Categories providing a proxy to discussion formats?

But the options are currently quite similar and limited at the moment. It would be nice to see a set of Discussion Formats that enable information to be created & communicated in a more diverse set of ways. An example of this could be the "WikiModes" used over at https://wiki.c2.com/?WikiModes. Whilst some of them are maybe too specific I could imagine the following being useful:

- [Thread Mode](https://wiki.c2.com/?ThreadMode) - Quite similar to the linearly - time sorted general discussions but can be shuffled and edited/refactored by people to improve the readability.
- [Opinion Mode](https://wiki.c2.com/?OpinionMode) - Similar to Question/Answer discussion with individuals allowed one editable entry (their opinion) which is appended underneath the "majority opinion".
- [Dialogue Mode](https://wiki.c2.com/?DialogMode) - I really like this one. Here there are two pseudonyms which contributors can post under to argue for/against some topic, effectively using the [dialectical method](https://en.wikipedia.org/wiki/Dialectic) in order to discover the best outcome or solution.
- [FAQ Mode](https://wiki.c2.com/?FaqMode) - Pretty self explanatory, but another useful format.
- [List Mode](https://wiki.c2.com/?ListMode) - Essentially a list of links or resources with a discussion underneath them.

There is also a [Document Mode](https://wiki.c2.com/?DocumentMode) but I would argue that discussions are transient tools enabling learning and shouldn't be focused on being used as Documentation.

## In Summary

1) So far, Discussions seem amazing.
2) We should use **labels** to organize/group discussions instead of added an unnecessary step of Categories.
   - (This reminds me of how much I loved moving to GitHub and simply creating "issues" instead of working out if I had a "task", or a "story", or a "spike", or a "subtask" etc....)
3) We could **extend the set of Discussion Formats** to allow for other environments to learn and debate. Discussion Formats could then replace what currently exists as Categories.
   - I'm sure certain Formats could also be converted from one to another (just like you can lock and unlock a discussion). 

I'm curious to see what other's thoughts are about this! What formats of Discussion have you found helpful?

**Comment by BekahHW** [comment]
I'm running community conversations through GH discussions, so having categories is really important to me. And then I use labels to further identify what the discussion is about.

For example, we have a monthly challenge team. They get a category, and then each post will get a label with the challenge. In another organization, I have the Show & Tell category, and we use a label that's show and tell that uses our product and another label is for personal project. 

Can you use labels instead of the categories? Absolutely, but it complicates searchability, ease of information access, and organization.
